.github/workflows/docs.yml

```diff
@@ -1,16 +1,27 @@
-name: Build docs on push
+name: Build and Publish Documentation
 
-on: [push]
+on: 
+  push:
+    paths:
+      - 'docs/*'
+      - '*.md'
+  pull_request:
+    paths:
+      - 'docs/*'
+      - '*.md'
+  workflow_dispatch:
+  create:
```

Comment:
I like the workflow_dispatch trigger, didn't know about that one! I'm not sure about create though. Do we really want to run it for every branch created?
I think the paths restrictions are too strict, they'd need to also include the configuration files, docs/document.py, and the contents of spec/, and now that's pretty much all of the files!

Maybe just 

```yaml
on:
  push:
  pull_request:
  workflow_dispatch:
```

Reply:
I dropped the path search but kept 'create' b/c I think we want it to create for branches and tags for now...we can drop if/when it starts being too much.

Note that the delete function will delete the docs for any reference (branch/tag) that we delete from the repo so we shouldn't get too crowded.